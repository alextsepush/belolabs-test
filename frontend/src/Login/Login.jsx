import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import axios from 'axios';
import backgroundGif from "../assets/images/play.gif";

const Login = ({ onLogin }) => {
  const [formData, setFormData] = useState({ username: '', password: '' });
  const [error, setError] = useState('');
  const navigate = useNavigate(); // For navigation

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const response = await axios.post('http://localhost:3000/api/users/login', formData);
      localStorage.setItem('token', response.data.token);
      localStorage.setItem('userID', response.data.userID); // Save the userID
      onLogin();
      navigate('/play');
    } catch (error) {
      if (error.response && error.response.status === 400) {
        setError('User not found. Please register first.');
      } else {
        setError(error.response?.data.message || 'Error logging in');
      }
    }
  };


  const handleRegisterRedirect = () => {
    navigate('/register'); // Redirect to the Register page
  };

  return (
      <div className="login-page" style={{
          backgroundImage: `url(${backgroundGif})`,
      }}>
          <h1 className={`game-title`}>
              WonderCards
          </h1>
    <form onSubmit={handleSubmit} className="login-box">
      <h2>Login</h2>
        <div className="login-inputs">
      <input
        type="text"
        placeholder="Username"
        value={formData.username}
        onChange={(e) => setFormData({ ...formData, username: e.target.value })}
      />
      <input
        type="password"
        placeholder="Password"
        value={formData.password}
        onChange={(e) => setFormData({ ...formData, password: e.target.value })}
      />
        </div>
      <div style={{ display: 'flex', gap: '10px' }}>
        <button data-testid="login-button" type="submit" disabled={!formData.password || !formData.username}>Login</button>
        <button data-testid="register-button" type="button" onClick={handleRegisterRedirect}>
          Register
        </button>
      </div>
      <p style={{ color: 'red' }}>{error}</p>
    </form>
      </div>
  );
};

export default Login;
