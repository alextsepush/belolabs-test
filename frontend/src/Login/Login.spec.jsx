import { render, screen } from "@testing-library/react";
import Login from "./Login";

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => () => null,
}));

describe("Login.spec.jsx", () => {
    it("Should render the login form with disabled login button if there are no values", () => {
        const { container} = render(<Login />);
        expect(container.firstChild).toMatchSnapshot();

        expect(screen.getByTestId('login-button')).toBeDisabled()
        expect(screen.getByTestId('register-button')).not.toBeDisabled()
    });
});
