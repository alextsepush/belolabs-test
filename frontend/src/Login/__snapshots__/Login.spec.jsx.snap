// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.spec.jsx Should render the login form 1`] = `
<div
  class="login-page"
  style="background-image: url(test-file-stub);"
>
  <h1
    class="game-title"
  >
    WonderCards
  </h1>
  <form
    class="login-box"
  >
    <h2>
      Login
    </h2>
    <div
      class="login-inputs"
    >
      <input
        placeholder="Username"
        type="text"
        value=""
      />
      <input
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <div
      style="display: flex; gap: 10px;"
    >
      <button
        data-testid="login-button"
        disabled=""
        type="submit"
      >
        Login
      </button>
      <button
        data-testid="register-button"
        type="button"
      >
        Register
      </button>
    </div>
    <p
      style="color: red;"
    />
  </form>
</div>
`;

exports[`Login.spec.jsx Should render the login form with disabled login button if there are no values 1`] = `
<div
  class="login-page"
  style="background-image: url(test-file-stub);"
>
  <h1
    class="game-title"
  >
    WonderCards
  </h1>
  <form
    class="login-box"
  >
    <h2>
      Login
    </h2>
    <div
      class="login-inputs"
    >
      <input
        placeholder="Username"
        type="text"
        value=""
      />
      <input
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <div
      style="display: flex; gap: 10px;"
    >
      <button
        data-testid="login-button"
        disabled=""
        type="submit"
      >
        Login
      </button>
      <button
        data-testid="register-button"
        type="button"
      >
        Register
      </button>
    </div>
    <p
      style="color: red;"
    />
  </form>
</div>
`;
